### PR TITLE
Make sure you can delete add-ons with unknown types

### PR DESCRIPTION
**Pull request by diox**
Unblocks deleting add-ons not compatible with Firefoxes, since
we have some webapps that are still in the database...

Hopefully fix https://github.com/mozilla/addons-server/issues/8645

### CODE REVIEW

File: src/olympia/addons/tests/test_models.py [hunk 1]
```diff
@@ -655,6 +655,14 @@ def test_delete_disabled_addon_when_guid_is_already_in_deniedguids(self):
         self._delete(3615)
         assert DeniedGuid.objects.filter(guid=addon.guid).exists()
 
+    def test_delete_unknown_type(self):
+        """
+        Test making sure deleting add-ons with an unknown type, like old
+        webapps from Marketplace that are somehow still around, is possible."""
+        addon = Addon.objects.get(pk=3615)
+        addon.update(type=11)
+        self._delete(3615)
```
Comment (eviljeff) [inline src/olympia/addons/tests/test_models.py]: Should we force hard-delete any unknown type?  Soft-deleted addons still hang around, causing weird edge cases.

Comment (diox) [reply src/olympia/addons/tests/test_models.py]: I didn't want to make it mandatory just because there was a weird type. I think it's best to force delete ourselves if we truly want to (something we might do in a few months for all non-Firefox add-ons as well)
